src/app/modules/issue/issue.controller.ts

```diff
@@ -0,0 +1,100 @@
+import { celebrate, Joi, Segments } from 'celebrate'
+import { StatusCodes } from 'http-status-codes'
+import { ErrorDto, PrivateFormErrorDto } from 'shared/types'
+
+import { createLoggerWithLabel } from '../../config/logger'
+import { createReqMeta } from '../../utils/request'
+import { ControllerHandler } from '../core/core.types'
+import { PrivateFormError } from '../form/form.errors'
+import * as FormService from '../form/form.service'
+
+import { insertFormIssue, notifyFormAdmin } from './issue.service'
+import { mapRouteError } from './issue.util'
+
+const logger = createLoggerWithLabel(module)
+
+const validateSubmitFormIssueParams = celebrate({
+  [Segments.BODY]: Joi.object().keys({
+    issue: Joi.string().trim().required(),
+    email: Joi.string()
+      .trim()
+      .email()
+      .message('Please enter a valid email')
+      .lowercase()
+      .allow(''),
+  }),
+})
+
+/**
+ * Handler for POST api/v3/forms/:formId/issue endpoint
+ * @precondition formId should be present in req.params.
+ * @precondition Joi validation should enforce shape of req.body before this handler is invoked.
+ *
+ * @returns 200 if feedback was successfully saved
+ * @returns 404 if form with formId does not exist or is private, or submissionId does not exist
+ * @returns 410 if form has been archived
+ * @returns 500 if database error occurs
+ */
+const submitFormIssue: ControllerHandler<
+  { formId: string },
+  { message: string } | ErrorDto | PrivateFormErrorDto,
+  { issue: string; email: string }
+> = async (req, res) => {
+  const { formId } = req.params
+  const { issue, email } = req.body
+  const logMeta = {
+    action: 'submitFormIssue',
+    ...createReqMeta(req),
+    formId,
+  }
+
+  return FormService.retrieveFormKeysById(formId, [
+    '_id',
+    'admin',
+    'inactiveMessage',
+    'permissionList',
+    'status',
+    'title',
+  ])
+    .andThen((form) => FormService.isFormPublic(form).map(() => form))
```

Comment:
Completely unrelated to the changes on this PR. 

This `FormService.isFormPublic(form)` is rather idiosyncratic. It returns `ok(true)/err(false)` which is a tautology.

To check if the value is false the caller wouldn't need to access the inner value:
```js
const result = FormService.isFormPublic(form);
if(result.isErr()) {
  // don't really need to consume the actual response
}

if(result.isTrue()) {
  // don't really need to consume the actual response either 🤔 
}
```

This causes the caller to require some gymnastics to chain the function like what it is done here instead of being able to directly use the result. 

With a small change like so:
```diff
// update isFormPublic to return `form`
export const isFormPublic = (
  form: IPopulatedForm,
): Result<true, FormDeletedError | PrivateFormError | ApplicationError> => {
  if (!form.status) {
    return err(new ApplicationError())
  }
  switch (form.status) {
    case FormStatus.Public:
-      return ok(true)
+      return ok(form)
    case FormStatus.Archived:
      return err(new FormDeletedError())
    case FormStatus.Private:
      return err(new PrivateFormError(form.inactiveMessage, form.title))
  }
}
```

We can directly chain the result:
```js
FormService.retrieveFormKeysById(...args)
  .andThen(FormService.isFormPublic)
  .andThen(doSomething)
```

Wondering if you also had the same observation when working on this.

Reply:
Cmiiw but currently the error is being consumed, to report why is the form not public (tho i doubt we should be doing this, for security reason). 

As for the argument of returning true/false vs the form object, I think since the function name is isXXXX, naturally, caller will expect it to function as a predicate function, which also translates to better readability. But since we are already returning an error, this is no longer appropriate for a predicate function, which means returning the form object itself will makes it easier for us to chain. Also, in term of naming, perhaps something like validatePublicForm will be better to avoid misunderstanding and assumption that the function will return a boolean type.

In summary, I agree with your point, but i will keep this here for now, and we can do some refactoring in the future. Lmk if you think otherwise.

Reply:
👍 Yeah definitely not suggesting refactoring here. ❌ feature creeps!